app/utils/get_filtered_records.py

```diff
@@ -221,10 +219,16 @@ def date_filter(record, start_date=None, end_date=None, use_sampling_date=True):
                 # True/False have dtype="object" instead of "bool"
                 if True in prioritized_records[col].values:
                     prioritized_records[col] = prioritized_records[col].fillna(False)
-            # Filling all NaN values with None: https://stackoverflow.com/questions/46283312/how-to-proceed-with-none-value-in-pandas-fillna
             prioritized_records = prioritized_records.fillna(np.nan).replace({np.nan: None})
         result = prioritized_records.to_dict('records')
 
+    # If estimates_subgroup is 'primary_estimates', just return the primary estimate for each study
```

Comment:
I think this would be more performant to do this filtering step right after `get_all_records` is called (and before all other filtering happens)